test/functional/mochitest/browser_as_private_browsing.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+let Cu = Components.utils;
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+/**
+ * Tests that loading activity streams in private browsing mode doesn't actually
+ * load the activity streams URL and instead loads about:privatebrowsing
+ */
+add_task(function*doesnt_load_in_private_windows() {
+  const asURL = "resource://activity-streams/data/content/activity-streams.html#/";
+  const privateBrowsingURL = "about:privatebrowsing";
+
+  // create a private browsing window
+  let privateWindow = yield BrowserTestUtils.openNewBrowserWindow({private: true});
+  let tab = privateWindow.gBrowser.selectedBrowser;
+
+  // try to load activity stream's url into the private window
+  tab.loadURI(asURL);
+  yield BrowserTestUtils.browserLoaded(tab);
+
+  // check what the content task thinks has been loaded
+  yield ContentTask.spawn(tab, {url: privateBrowsingURL}, args => {
+    Assert.equal(content.location.href, args.url, "Correctly loaded about:privatebrowsing instead of activity-streams");
```

Comment:
From poking around, it looks like content.location.href should reference the right thing, but I haven't yet found where that's documented or implemented.  Do you happen to know?